### ISSUE TITLE
Bad results when training Transformer TTS from EGS2 LJSpeech recipe

### ISSUE THREAD

**Issue by napalm00** [espnet/espnet#2565]
I'm experimenting with training LJSpeech Transformer TTS from the EGS2 recipe in the repo.

I'm using a custom dataset based on audios from Youtube (Donald Trump voice, as it's the most readily available) for a total dataset duration of 8 hours. The audios are clean of noise and split into 6-12 second segments, much like LJSpeech.
The text has been obtained through Google Speech-to-Text and numbers/abbreviations have been expanded into words. I'm attaching a few samples of the dataset for reference.

I've experimented with both fine tuning from a pre-trained Transformer model (from model zoo, as described [here](https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/tts1)) and training from scratch.
I'm unable to achieve any decent speech when decoding, with the voice being very robotic and making no sense at all when speaking.

I've looked at the train charts for both the pre-trained LJSpeech Transformer TTS and my own results, and noticed that the bce_loss and decoder_alpha charts look very different, with the "valid" curve increasing in my dataset instead of decreasing like in LJSpeech. 

I've trained both the LJSpeech dataset and my dataset from scratch (using the same config) for 12hrs on a 4x Tesla V100 GPU instance, and compared the results; I'm attaching both charts folders and also a direct comparison between those two charts.

![comparison](https://user-images.githubusercontent.com/4140600/95574905-0bf04080-0a2e-11eb-93f5-2ea2bdb05928.png)

As far as I can see, there's little to no difference between my dataset's audios/text and LJSpeech's. 
Clearly it's something with my dataset, however I can't put my finger on what :confused: 

[trump_sample.zip](https://github.com/espnet/espnet/files/5354404/trump_sample.zip)

[ljspeech_charts.zip](https://github.com/espnet/espnet/files/5354401/ljspeech_charts.zip)
[custom_charts.zip](https://github.com/espnet/espnet/files/5354400/custom_charts.zip)

**Comment by kan-bayashi** [comment]
First, please check FAQ:
https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/tts1#why-the-model-cannot-be-trained-well-with-my-own-dataset

In the AR E2E-TTS model, the attention is the most important, the loss value is not so meaningful.
Please check the attention plot is diagonal (saved in `exp/*/att_ws/`).
Maybe the problem is the silence which does not correspond to the silence.
This makes the diagonal attention learning difficult.

**Comment by napalm00** [comment]
The attention plot seems to be all over the place for my dataset (epoch 52):

![trump_tts decoder decoders 0 src_attn 52ep](https://user-images.githubusercontent.com/4140600/95581725-c9ccfc00-0a39-11eb-9d36-709c7bc9f966.png)

compared to LJSpeech (also epoch 52):

![ljspeech_tts decoder decoders 0 src_attn 52ep](https://user-images.githubusercontent.com/4140600/95581758-d94c4500-0a39-11eb-9011-d3a1a0823235.png)

> Maybe the problem is the silence which does not correspond to the character.

Do you mean there should be a period/comma in the transcription whenever there's a silence break?
I've tried to model my dataset on the basis of LJSpeech; I've read the FAQ but I can't really spot any weird silences/long pauses in my data, the longest pause between two sentences in my dataset is 300ms - is this too long?
Spectrogram comparison:
![spectrogram](https://user-images.githubusercontent.com/4140600/95581980-3647fb00-0a3a-11eb-9239-58131a4de1dc.png)

**Comment by kan-bayashi** [comment]
> Do you mean there should be a period/comma in the transcription whenever there's a silence break?

Yes. It is better to add such a symbol.
For example, during 300 ms / (256 shfit pt / 22050 hz * 1000) = 26 frames, the model cannot decide where should be focused. This makes the attention learning very difficult.
In my case, I always remove silence at the beginning and the end of the audio using force alignment results and use a comma-like symbol for silence in the middle of the audio.

You may be able to train the model with a large reduction factor (say `reduction_factor=5`) and a large guided attention lambda.

**Comment by napalm00** [comment]
Fantastic, thank you for the detailed reply! I'll fix the dataset as you said and report back with the results.

**Comment by napalm00** [comment]
Can confirm that the pauses and lack of commas were indeed the problem, I'm seeing better results across the board now!
Thank you for your help! I'll close the issue as it's now solved.

**Comment by Curiosci** [comment]
Hi! Could you please provide more details on how you modified your dataset transcriptions? what silence length did you consider to add commas in the training data? I
